notmuch/db.c

```diff
@@ -102,6 +103,18 @@ notmuch_database_t *nm_db_do_open(const char *filename, bool writable, bool verb
   const notmuch_database_mode_t mode =
       writable ? NOTMUCH_DATABASE_MODE_READ_WRITE : NOTMUCH_DATABASE_MODE_READ_ONLY;
 
+  struct stat st1;
+  struct stat st2;
+  char buf[PATH_MAX];
+  if (stat(filename, &st1) != 0)
+    mutt_error(_("Can't stat %s: %s"), filename, strerror(errno));
+  else if (!S_ISDIR(st1.st_mode))
+    mutt_error(_("%s is not a directory"), filename);
+  if (stat(mutt_path_concat(buf, filename, ".notmuch", sizeof(buf)), &st2) != 0)
+    mutt_error(_("Can't stat %s: %s"), filename, strerror(errno));
+  else if(!S_ISDIR(st2.st_mode))
+    mutt_error(_("%s is not a directory"), buf);
```

Comment:
Whenever I hit this conditional, NeoMutt locks up and must be killed externally. It appears to be in an infinite loop of trying to open the database, hitting this conditional, then trying again.

Here's a small script to reproduce it. Just change the paths accordingly.
```
mkdir -p /home/aray/tmp
touch /home/aray/tmp/.notmuch
echo 'set nm_default_uri="notmuch:///home/aray/tmp"' >>.neomuttrc.test
echo 'set spoolfile="notmuch://?query=tag:inbox"' >> .neomuttrc.test
./neomutt -n -F .neomuttrc.test
```

Reply:
I've added some `return`s on the error paths.
NeoMutt doesn't lock up using your example any more.

Reply:
Awesome! I'll mark this as resolved.